### ISSUE TITLE
Production build fails but dev succeeds when module is amd

### ISSUE THREAD

**Issue by HarelM** [angular/angular-cli#8701]
### Versions
```
Output from: `ng --version`.
Angular CLI: 1.5.5
Node: 8.9.1
OS: win32 x64
Angular: 5.0.3
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 5.0.0-rc.2
@angular/cli: 1.5.5
@angular/flex-layout: 2.0.0-beta.10-4905443
@angular/material: 5.0.0-rc.2
@angular-devkit/build-optimizer: 0.0.34
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.38
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.5
@schematics/angular: 0.1.8
typescript: 2.4.2
webpack: 3.8.1
```

### Repro steps
clone project from https://github.com/IsraelHikingMap/Site/
run `ng build --prod`

### Observed behavior
compilation succeeds on dev but fails on production
```
ERROR in ./sources/application/application.module.ngfactory.js
  Module parse failed: Unexpected token (1:4867)
  You may need an appropriate loader to handle this file type.
  | define(["require", "exports", "@angular/core", "./application.module", "./components/main-map.component", "./components/sidebar/sidebar.component", "./components/route-statistics-chart.component", "../../node_modules/@angular/material/dialog/typings/index.ngfactory", "../../node_modules/@angular/material/snack-bar/typings/index.ngfactory", "../../node_modules/@angular/material/tooltip/typings/index.ngfactory", "./components/zoom.component.ngfactory", "./components/location-button.component.
...
```

### Desired behavior
Better error message or compilation success

### Mention any other details that might be useful (optional)
Probably a dup, but couldn't find the exact issue.
This ran with success with 1.4.4 and angular 4.4.6
Relevant CI run can be found here:
https://ci.appveyor.com/project/IsraelHikingHost/site/build/5.2.43

**Comment by HarelM** [comment]
After further trail and error today the issue was caused due to the following typescript change:
In typescript 2.3 in `tsconfig.app.json` `module` was `es2016`.
Upgrading to 2.4 caused the following error:
```
error TS6046: Argument for '--module' option must be: 'none', 'commonjs', 'amd', 'system', 'umd', 'es6', 'es2015', 'esnext'.
```
While trying to solve this I changed `es2016` to `amd` (a post somewhere said this was the way to solve this).
This solved the dev compilation but caused a different error in aot compilation.
So, to summarize - the solution is to use `esnext` in `module`:
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "module": "esnext",
    "baseUrl": "",
    "types": [
      "google.visualization"
    ]
  },
  "include": [
    "**/*.ts"
  ],
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ]
}
```
I'll change the issue's title.

**Comment by filipesilva** [comment]
This does seem like a hard-to-debug issue but ultimately it was because of that AMD change. I'm closing it because I don't think there's much we can do.

**Comment by HarelM** [comment]
Please consider reopening this issue, as I said the following is what I expect:
>Better error message or compilation success

The fact that dev succeeds and production fails without a clear error message almost caused me to downgrade back to cli 1.4 as I couldn't find what I did wrong...
